### PR TITLE
Use more perl signatures

### PR DESCRIPTION
**Pull request by okurz**
* Use proper perl "signatures" in common files - osutils

### CODE REVIEW

**Comment by okurz** [comment]
As discussed in https://progress.opensuse.org/issues/99579 and as planned in https://progress.opensuse.org/issues/99663 we should use more perl signatures. I had started already here but as I recurringly fail to finish I am splitting this up in multiple PRs for easier testing and planned merges one at a time:

* https://github.com/os-autoinst/os-autoinst/pull/1789
* https://github.com/os-autoinst/os-autoinst/pull/1790
* https://github.com/os-autoinst/os-autoinst/pull/1791
* https://github.com/os-autoinst/os-autoinst/pull/1792
* https://github.com/os-autoinst/os-autoinst/pull/1793
* https://github.com/os-autoinst/os-autoinst/pull/1794

let's see which ones succeed. Here I have that really weird CI running into timeout after 360m

**Comment by okurz** [comment]
https://build.opensuse.org/build/devel:openQA:GitHub:os-autoinst:os-autoinst:PR-1773/openSUSE_Tumbleweed/x86_64/os-autoinst/_log says multiple "Syntax check" failures in the Tumbleweed based checks though these are not seen in GHA tests and not in the Leap variants. Maybe Tumbleweed has a newer, stricter version of perlcritic?

**Comment by okurz** [comment]
> https://build.opensuse.org/build/devel:openQA:GitHub:os-autoinst:os-autoinst:PR-1773/openSUSE_Tumbleweed/x86_64/os-autoinst/_log says multiple "Syntax check" failures in the Tumbleweed based checks though these are not seen in GHA tests and not in the Leap variants. Maybe Tumbleweed has a newer, stricter version of perlcritic?

I managed to scroll up and found the message which tells that prototypes should come first. Let's see if both OS and perl versions support that.

**Comment by okurz** [comment]
codecov/patch isn't met but it's partially in deprecated backends which we don't want to care about much anyway so I would say this is ok.

File: osutils.pm [hunk 1]
```diff
@@ -52,30 +43,23 @@ sub gen_params(\@$$;%) {
...
 # Add single quote mark to string
 # Mainly use in the case of multiple kernel parameters to be passed to the -append option
 # and they need to be quoted using single or double quotes
-sub quote {
-    "\'" . $_[0] . "\'";
-}
-
-sub run {
-    my @cmd = @_;
+sub quote ($string) { "\'" . $string . "\'" }
```
[the remark this answers is not in the document]
Comment (okurz) [reply osutils.pm]: ok, it's an unrelated change so I added a separate commit

**Review by Martchus — APPROVED** [review]

**Review by perlpunk — APPROVED** [review]

**Pull request merged**